src/main/java/org/broadinstitute/dsm/model/ParticipantWrapper.java

```diff
@@ -124,7 +125,15 @@ else if (StringUtils.isNotBlank(instance.getParticipantIndexES())){
             }
             //get all the list which were not filtered
             if (participantESData == null) {
-                participantESData = getESData(instance);
+                //get only pts for the filtered kitRequests
+                if (kitRequests != null && !kitRequests.isEmpty()) {
```

Comment:
This looks like it will help.  `KitRequestShipping.getKitRequests()` on L109 is still an issue, however.  It looks like that method will still pull in all kit requests for the study.  Is it possible

Reply:
I think we may have to do something like set `limit 1000` and order by something like `user_id desc` in `KitRequestShipping.getKitRequests()` when there are no filters passed in.  Otherwise, we are going to read in everything into memory.  It's a drastic limitation, but short of rearchitecting things so that **all** data that feeds these views goes into ES and queries **only** stream out of ES (avoiding app server memory entirely with either UI redirects or some such), I don't see an elegant or quick way to ensure that DSM won't blow out memory limits.

Reply:
Come to think of it, maybe we should apply this kind of limit across the board until we come up with something better--no matter what, we cap the size of results at 1k?  5k?

Reply:
I will add the filtering to L109 and L66 if the list was filtered for something and if not I will add a default cap of results and a comment to the UI that if something is missing user should filter for that pt to get "all" informations